posix.mak

```diff
@@ -578,10 +578,19 @@ dpl-docs: ${DUB} ${STABLE_DMD}
 	DFLAGS="$(DPL_DOCS_DFLAGS)" ${DUB} build --root=${DPL_DOCS_PATH} \
 		--compiler=${STABLE_DMD}
 
-${STABLE_DMD}:
-	mkdir -p ${STABLE_DMD_ROOT}
+# .tar.xz is archives are smaller (and don't need a temporary dir) -> prefer if available
+HAS_XZ := $(shell command -v xz 2> /dev/null)
+${STABLE_DMD_ROOT}/.downloaded:
+	@mkdir -p $(dir $@)
+ifdef HAS_XZ
```

Comment:
I'd say that always running a shell command to define a global make variable (HAS_XZ) is a bit of an anti-pattern.
Might be nicer to just do the if/else in the shell, also provide more helpful output.
```make
    if command -v xz 2>/dev/null; then \
        curl -fSL --retry 3 $(subst .zip,.tar.xz,$(STABLE_DMD_URL)) | tar -Jxf - -C $(dir $@); \
    else \
        TMPFILE=$$(mktemp deleteme.XXXXXXXX) && curl -fsSL ${STABLE_DMD_URL} > ${TMP}/$${TMPFILE}.zip && \
            unzip -qd ${STABLE_DMD_ROOT} ${TMP}/$${TMPFILE}.zip && rm ${TMP}/$${TMPFILE}.zip; \
    fi
```
Just make sure to use semicolons (`;`) to end commands as `\` in make will not result in a newline, so the shell sees the command in a single line and needs `;` separators `if exp; then exp; else exp; fi`.